### PR TITLE
Putting geocoder methods back

### PR DESCRIPTION
**Pull request by hanbyul-here**
- I dropped some methods that mapzen.js offered when we change the way of bundling geocoder. Putting those back
- I added the tests just to check there are these methods. (not to make this kind of mistake again) 👈 I took these out since there is a mystery thing going on from Leaflet side breaking tests.

### CODE REVIEW

File: src/js/components/search.js [hunk 1]
```diff
@@ -1,5 +1,23 @@
...
+
+Geocoder.prototype.getSearchResult = function (input, callback) {
+  var param = {
+    text: input
+  };
+  var params = this.getParams(param);
+  corslite(this.options.url + '/search?' + this.serialize(params), callback, true);
+};
+
+Geocoder.prototype.getAutoCompleteResult = function (input, callback) {
```
Comment (louh) [inline src/js/components/search.js]: Looks like the camelcasing here is different from the [documentation](https://mapzen.com/documentation/mapzen-js/search/#methods): (`getAutoCompleteResult ` vs `getAutocompleteResult`)

Comment (hanbyul-here) [reply src/js/components/search.js]: arg! Thanks for pointing this!

Comment (louh) [reply src/js/components/search.js]: just a thought, does changing this fix any of the tests?

Comment (hanbyul-here) [reply src/js/components/search.js]: My tests were having same problems that you had (it passes on local, but not on Circle!! Arg!) I will push the test commit.

Comment (hanbyul-here) [reply src/js/components/search.js]: No it didn't :( I will take out the tests

Comment (louh) [reply src/js/components/search.js]: really? I thought this was related :-/

```
PhantomJS 2.1.1 (Linux 0.0.0) Search Test Basic Geocoder Check checks geocoder autocomplete methods without UI exists. FAILED
	undefined is not a function (evaluating 'geocoder.getAutocompleteResult('dummy', function (err, resp) {
	        done();
	      })')
	/tmp/7a77da693d622ed055d5991f2e29270b.browserify:213:37 <- test/spec/search.js:39:37
```

ok

Comment (hanbyul-here) [reply src/js/components/search.js]: yes it did solve that problem but it rasied the other problem that `getSearchResult` method was throwing. (https://circleci.com/gh/mapzen/mapzen.js/1262) 
```
undefined is not an object (evaluating 'this._map.getCenter')
```

**Pull request merged**